ethers-ens/src/main/kotlin/io/ethers/ens/EnsResolver.kt

```diff
@@ -0,0 +1,442 @@
+package io.ethers.ens
+
+import io.ethers.abi.AbiCodec
+import io.ethers.abi.AbiType
+import io.ethers.core.FastHex
+import io.ethers.core.Jackson
+import io.ethers.core.types.Address
+import io.ethers.core.types.BlockId
+import io.ethers.core.types.Bytes
+import io.ethers.core.types.CallRequest
+import io.ethers.logger.err
+import io.ethers.logger.getLogger
+import io.ethers.logger.wrn
+import io.ethers.providers.Provider
+import io.ethers.providers.types.RpcResponse
+import io.github.adraffy.ens.InvalidLabelException
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import java.io.IOException
+import java.util.concurrent.CompletableFuture
+
+class EnsResolver(
+    private val provider: Provider,
+    private val registryAddress: Address,
+    private val ccipLookupLimit: Int = 4,
+    private val client: OkHttpClient = OkHttpClient(),
+) {
+    private val LOG = getLogger()
+    constructor(provider: Provider) : this(provider, getRegistryAddressOrThrow(provider.chainId))
```

Comment:
Add all parameters from primary constructor - except `registryAddress` - to this constructor as well, including default values. And annotate both of them with `@JvmOverloads` so it becomes more ergonomic to use from java.